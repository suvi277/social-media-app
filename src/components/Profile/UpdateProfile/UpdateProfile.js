import React, { useState } from 'react';
import { TextField, Button } from '@app/components/UI';

const UpdateProfile = ({ id, firstName, lastName, email, updateProfile, cancelEditing }) => {
	const [ currentUser, setCurrentUser ] = useState({
		id,
		firstName,
		lastName,
		email
	});

	const handleChange = (e) => {
		const { name, value } = e.target;
		setCurrentUser((user) => ({ ...user, [name]: value }));
	};

	const updateClicked = (e) => {
		updateProfile(currentUser);
	};

	return (
		<div className="w-50">
			<TextField
				label="First Name"
				name="firstName"
				type="text"
				onChange={handleChange}
				value={currentUser.firstName}
				className="input"
			/>
			<TextField
				label="Last Name"
				name="lastName"
				onChange={handleChange}
				type="text"
				value={currentUser.lastName}
				className="input"
			/>
			<TextField
				label="Email"
				onChange={handleChange}
				name="email"
				type="email"
				value={currentUser.email}
				className="input"
			/>
			<Button type="button" label="Update" className="btn-primary mr-3" buttonClicked={updateClicked} />
			<Button type="button" label="Cancel" className="btn-outline-primary" buttonClicked={cancelEditing} />
		</div>
	);
};

export { UpdateProfile };
