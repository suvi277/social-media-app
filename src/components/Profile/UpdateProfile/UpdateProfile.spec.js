import React from 'react';
import { shallow } from 'enzyme';
import { UpdateProfile } from './UpdateProfile';

describe('UpdateProfile Component', () => {
	const wrapper = shallow(<UpdateProfile />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	//To Complet Test Cases
});
