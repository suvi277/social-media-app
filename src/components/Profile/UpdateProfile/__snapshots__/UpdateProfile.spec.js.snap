// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateProfile Component renders as expected 1`] = `ShallowWrapper {}`;
